packages/@uppy/companion/src/server/controllers/s3.js

```diff
@@ -353,7 +357,60 @@ module.exports = function s3 (config) {
     }, next)
   }
 
+  const policy = {
+    Version: '2012-10-17',
+    Statement: [
+      {
+        Effect: 'Allow',
+        Action: [
+          's3:PutObject',
+        ],
+        Resource: [
+          `arn:aws:s3:::${config.bucket}/*`,
+          `arn:aws:s3:::${config.bucket}`,
+        ],
+      },
+    ],
+  }
+
+  let stsClient
+  function getSTSClient () {
+    if (stsClient == null) {
+      stsClient = new STSClient({
+        region: config.region,
+        credentials : {
+          accessKeyId: config.key,
+          secretAccessKey: config.secret,
+        },
+      })
+    }
+    return stsClient
+  }
+
+  function getTemporarySecurityCredentials (req, res, next) {
+    getSTSClient().send(new GetFederationTokenCommand({
+      Name: '123user',
```

Comment:
123user doesn't sounds like something everyone would want. Should we remove this parameter, or allow customising it? or if it doesn't matter what `Name` is, maybe add a comment about that.

Reply:
I'm not sure about this, I think it would be useful in some situation to differentiate users (e.g. to understand where a spike of usage comes from), not sure how we would make that happen. Maybe we should just name if `'companion'` and if users want something else, they need to implement it themselves?

Reply:
sounds good